extensions/amp-access/0.1/amp-access-server.js

```diff
@@ -180,6 +193,39 @@ export class AccessServerAdapter {
     return this.clientAdapter_.pingback();
   }
 
+  /** @override */
+  encodeReaderId(readerId) {
+    return this.getSubtleWithKey_().then(key => {
+      return this.subtle_.encrypt(key, readerId).then(s => {
+        dev.error(TAG, "bernie: encodeReaderId(): ", s);
+      });
+    });
+  }
+
+  /**
+   * @return {!Promise<!CryptoKey>}
+   * @private
+   */
+  getSubtleWithKey_() {
+    if (!this.keyPromise_) {
+      if (this.win.crypto) {
+        this.subtle_ =
+            this.win.crypto.subtle || this.win.crypto.webkitSubtle || null;
+      }
+      if (!this.subtle_) {
+        this.keyPromise_ = Promise.reject(new Error('No WebCrypto'));
+      } else {
+        this.keyPromise_ = this.subtle_.importKey(
+            'jwk',
+            'public_key: {"alg":"RSA-OAEP-256","e":"AQAB","ext":true,"key_ops":["encrypt"],"kty":"RSA","n":"nNfur__0p5qW_fGaTW4BNvtqx8SVxVfE3EkY78SS4AT68vjPq1kL82FMmt2Un3FPX_BAD22NB2LVziT3PaweWhgitbyv-JMjVymnza0LRETq2j-AlmETEpcHKtVSO70axZBzj8A0kIx_5d_ZSaScDYstVcWlSOfDh0575N-v4wWj5gyCSjIY0AJlmqToU2wIHrVdQwrvCj-2m4K86GRf_UF8NCt0RntN4CfRCqihoSIRKxHz12-CiCKhFwB3JyLcpMmMy_68h3jSYUP_YrVUXCYKnNNWmfLSDrP00juKBaQgwzPhpS_hkRkHgw9MxUit7JgxJH8JmCv5zWH1B8Un3w"}',
```

Comment:
Alright. Let's read the key from `configJson` (passed to the constructor). If key was not passed - just return rejection:
```
...
} else if (!rawKey) {
  this.keyPromise_ = Promise.reject(new Error('No key in the config'));
}
```

You probably want to configure it like this:
```
configJson = {
  "format": 'jwk',
  "key": 'BLOB'
}
```

Other ways are possible, of course. So, suggest your own if you'd like.